rotkehlchen/chain/ethereum/modules/makerdao/sai/decoder.py

```diff
@@ -0,0 +1,521 @@
+                    location_label=liquidator,
+                    notes=f'Liquidate {amount} PETH for CDP {cdp_id}',
+                    counterparty=CPT_SAI,
+                )
+                already_decoded = True
+                return event, []
+
+        return None, []
+
+    def _decode_makerdao_deposit_weth_eth_event(
+            self,
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: Optional[list[ActionItem]],  # pylint: disable=unused-argument
+    ) -> tuple[Optional[HistoryBaseEntry], list[ActionItem]]:
+        owner_address = hex_or_bytes_to_address(tx_log.topics[1])
+        amount_raw = hex_or_bytes_to_int(tx_log.topics[2])
+        amount = asset_normalized_value(amount_raw, self.weth)
+
+        # check if ETH is sent to sai proxy contract or a proxy and mark it
+        # as a deposit since the proxy handles the conversion to weth
+        # and transfer to the sai tub contract
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount.is_close(amount, max_diff='0.005') is True
+            ):
+                event.balance.amount = amount
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Deposit {event.balance.amount} {self.eth.symbol} for PETH in SAI vault'  # noqa: E501
+                event.counterparty = CPT_SAI
+                break
+
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.counterparty == MAKERDAO_SAI_PROXY_CONTRACT and
+                event.asset == self.eth
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Deposit {event.balance.amount} {self.eth.symbol} for PETH in SAI vault'  # noqa: E501
+                event.counterparty = CPT_SAI
+                break
+
+        # NotERC20Conformant is never raised because PETH is a valid ERC20 token
+        get_or_create_evm_token(
+            userdb=self.base_tools.database,
+            chain_id=ChainID.ETHEREUM,
+            token_kind=EvmTokenKind.ERC20,
+            evm_address=POOLED_ETHER_ADDRESS,
+            evm_inquirer=self.ethereum_inquirer,
+        )
+
+        # the out event is already enriched or decoded above
+        in_event = HistoryBaseEntry(
+            event_identifier=transaction.tx_hash,
+            sequence_index=tx_log.log_index + 1000,  # force the event to come after the deposit
```

Comment:
This is really ugly. We have a function that swaps events order. Use that.

Reply:
This is still hacky and bad and won't fly. It's broken. It's your job to figure out a solution. If the deposit happens at a later point it's when the deposit comes that you have to remember this and make sure they come one after the other. Sending it "1000" events later is just silly and a hack.

The events need to appear consecutive, not just "at some point after" the other.

Otherwise accounting will not work.